packages/server/src/main/java/software/uncharted/terarium/hmiserver/models/dataservice/document/DocumentAsset.java

```diff
@@ -71,4 +97,39 @@ public static String getDocumentDoi(final Document doc) {
 		}
 		return docIdentifier;
 	}
+
+	@Override
+	public DocumentAsset clone() {
+		final DocumentAsset clone = new DocumentAsset();
+		super.cloneSuperFields(clone);
+
+		if (this.fileNames != null) {
+			clone.fileNames = new ArrayList<>(this.fileNames.size());
+			clone.fileNames.addAll(this.fileNames);
+		}
+
+		clone.documentUrl = this.documentUrl;
+
+		if (this.metadata != null) {
+			clone.metadata = new HashMap<>();
+			for (final String key : this.metadata.keySet()) {
+				// I don't like that this is an "object" because it doesn't clone nicely...
+				clone.metadata.put(key, this.metadata.get(key));
+			}
+		}
+
+		clone.source = this.source;
+		clone.text = this.text;
+
+		if (this.grounding != null) {
+			clone.grounding = this.grounding.clone();
+		}
+
+		if (this.assets != null) {
+			clone.assets = new ArrayList<>();
+			clone.assets.addAll(this.assets);
```

Comment:
Should this call the `clone` method on the DocumentExtractions?

Reply:
oh! thanks good catch